### PR TITLE
Allow emoji as link text

### PR DESCRIPTION
**Pull request by bernhardoj**
### Fixed Issues
$ https://github.com/Expensify/App/issues/44835

# Tests
1. What unit/integration tests cover your change? What autoQA tests cover your change?
Updated unit test.

1. What tests did you perform that validates your changed worked?
a. Send a link with emoji as text (`[😄 link](google.com)`) to any user
b. Verify `😄 link` is rendered as a link

Web
Screenshot 2024-07-11 at 12 07 30

Desktop
image

iOS mWeb
image

iOS
image

Android mWeb
image

Android
image

# QA
1. What does QA need to do to validate your changes?
1. What areas to they need to test for regressions?
Same as Test step above

### CODE REVIEW

**Comment by bernhardoj** [comment]
This PR only covers link. Do we also want to update it for email?
image

**Comment by bernhardoj** [comment]
@rayane-djouah updated

File: __tests__/ExpensiMark-HTML-test.js [hunk 1]
```diff
@@ -1200,7 +1200,7 @@ test('Test for link with no content', () => {
 
 test('Test for link with emoji', () => {
     const testString = '[😀](www.link.com)';
```
[the remark this answers is not in the document]
Comment (bernhardoj) [reply __tests__/ExpensiMark-HTML-test.js:1203]: Added as comment

File: __tests__/ExpensiMark-HTML-test.js [hunk 2]
```diff
@@ -304,12 +304,12 @@ test('Test markdown replacement for emojis with emails', () => {
         '[😄 abc@gmail.com ](abc@gmail.com) ';
     const result =
         'Do not replace the emoji with link ' +
```
[the remark this answers is not in the document]
Comment (bernhardoj) [reply __tests__/ExpensiMark-HTML-test.js]: Done

**Review by francoisl — APPROVED** [review]

**Pull request merged**